src/common/imageio.c

```diff
@@ -708,8 +708,10 @@ int dt_imageio_export_with_flags(const int32_t imgid, const char *filename,
 
     GList *modules_used = NULL;
 
-    dt_dev_pop_history_items_ext(&dev, dev.history_end);
-
+    // if we export with history overwrite we clear the complete stack
+    if(format_params->style_append  == FALSE)
+      dt_dev_pop_history_items_ext(&dev, dev.history_end);
```

Comment:
I don't think this will fix my issue. I've added a few test images in the issue #9191  to show the behavior I'm experiencing.